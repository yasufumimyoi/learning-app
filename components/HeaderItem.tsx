const HeaderItem = ({ Icon, title }: { Icon: any; title: string }) => {
  return (
    <div className="flex flex-col items-center group hover:opacity-50 duration-300">
      <Icon className="h-5 text-green-600 group-hover:animate-bounce mb-1" />
      <p className="text-xs">{title}</p>
    </div>
  )
}

export default HeaderItem
