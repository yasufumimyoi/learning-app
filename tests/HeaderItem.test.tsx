import { render, screen } from '@testing-library/react'
import HeaderItem from '../components/HeaderItem'
import { AcademicCapIcon } from '@heroicons/react/outline'

it('HeaderItemのsnpapshot"', () => {
  const component = render(<HeaderItem Icon={AcademicCapIcon} title="PROFILE" />)
  expect(component).toMatchSnapshot()
})

it('アイコンの文字表示"', () => {
  render(<HeaderItem Icon={AcademicCapIcon} title="PROFILE" />)
  const element = screen.getByText(/PROFILE/)
  expect(element).toBeInTheDocument()
})
