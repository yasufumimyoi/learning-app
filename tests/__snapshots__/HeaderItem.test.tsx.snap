// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderItem„ÅÆsnpapshot" 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col items-center group hover:opacity-50 duration-300"
      >
        <svg
          class="h-5 text-green-600 group-hover:animate-bounce mb-1"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 14l9-5-9-5-9 5 9 5z"
          />
          <path
            d="M12 14l6.16-3.422a12.083 12.083 0 01.665 6.479A11.952 11.952 0 0012 20.055a11.952 11.952 0 00-6.824-2.998 12.078 12.078 0 01.665-6.479L12 14z"
          />
          <path
            d="M12 14l9-5-9-5-9 5 9 5zm0 0l6.16-3.422a12.083 12.083 0 01.665 6.479A11.952 11.952 0 0012 20.055a11.952 11.952 0 00-6.824-2.998 12.078 12.078 0 01.665-6.479L12 14zm-4 6v-7.5l4-2.222"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <p
          class="text-white text-xs font-bold"
        >
          PROFILE
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col items-center group hover:opacity-50 duration-300"
    >
      <svg
        class="h-5 text-green-600 group-hover:animate-bounce mb-1"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 14l9-5-9-5-9 5 9 5z"
        />
        <path
          d="M12 14l6.16-3.422a12.083 12.083 0 01.665 6.479A11.952 11.952 0 0012 20.055a11.952 11.952 0 00-6.824-2.998 12.078 12.078 0 01.665-6.479L12 14z"
        />
        <path
          d="M12 14l9-5-9-5-9 5 9 5zm0 0l6.16-3.422a12.083 12.083 0 01.665 6.479A11.952 11.952 0 0012 20.055a11.952 11.952 0 00-6.824-2.998 12.078 12.078 0 01.665-6.479L12 14zm-4 6v-7.5l4-2.222"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
      <p
        class="text-white text-xs font-bold"
      >
        PROFILE
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
